src/PluginForm/Bluesnap/HostedPaymentFieldsPaymentMethodAddForm.php

```diff
@@ -135,6 +139,28 @@ protected function buildCreditCardForm(
       '#weight' => -200,
     ];
 
+    // Add BlueSnap JavaScript iFrame.
+    $url = 'https://www.bluesnap.com';
+    if ($this->entity->getPaymentGateway()->getPluginConfiguration() == "test") {
+      $url = 'https://sandbox.bluesnap.com';
+    }
+    $element['fraud_prevention'] = [
+      '#type' => 'inline_template',
+      '#template' => '
+        <iframe
+          width="1"
+          height="1"
+          frameborder="0"
+          scrolling="no"
+          src="{{ url }}/servlet/logo.htm?s={{ fraud_session_id }}">
+          <img width="1" height="1" src="{{ url }}"/servlet/logo.gif?s>
+        </iframe>',
+      '#context' => [
+        'url' => $url,
+        'fraud_session_id' => $this->fraudSession->get(),
+      ],
+    ];
```

Comment:
**CHANGE REQUEST**

You would want to refactor this to a separate function to avoid long functions, like we did with `ccFields/hiddenFields`. However, it's best to move this to the fraud session service, add it as a public `iframe($mode)` function.

Have you tested this though? Pretty sure the `getPluginConfiguration()` function does not return a string, it returns the configuration array. Best to do the following though:

```
$mode = $this->entity->getPaymentGateway()->getBluesnapConfig()['env'];
$this->fraudSession->iframe($mode);
```

Reply:
Made the change